### ISSUE TITLE
panic: runtime error: index out of range

### ISSUE THREAD

**Issue by metmajer** [hawkular/hawkular-openshift-agent#166]
I am using HOSA on an OpenShift Origin cluster using  `oc cluster up`:

```
$ oc version
oc v3.6.0-alpha.0+0343989
kubernetes v1.5.2+43a9be4
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://127.0.0.1:8443
openshift v3.6.0-alpha.0+0343989
kubernetes v1.5.2+43a9be4
```

With the following HOSA configuration:

```
kind: "ConfigMap"
apiVersion: "v1"
metadata:
  name: "api-gateway-hosa-config"
data:
  hawkular-openshift-agent: |
    endpoints:
    - type: "jolokia"
      url: "http://jolokia-api-gateway-helloworld.1.2.3.4.nip.io/jolokia/"
      credentials:
        token: "..."
      collection_interval: "30s"
      metrics:
      - name: "java.lang:type=Threading"
        type: "counter"
        id:   "VM Thread Count"
      - name: "java.lang:type=Memory"
        type: "gauge"
        id:   "VM Heap Memory Used"
```

, which is mounted into the Pod like so:

```
…
spec:
  containers:
  - name: "api-gateway"
    image: "metmajer/redhatmsa-api-gateway"
    env:
    - name: "AB_ENABLED"
      value: "jolokia"
    ports:
    - containerPort: 8080
      protocol: "TCP"
    - containerPort: 8778
      protocol: "TCP"
  volumes:
  - name: "hawkular-openshift-agent"
    configMap:
      name: "api-gateway-hosa-config"
```

However, when HOSA is started, it keeps crashing with said error:

```
]], config-map-volumes=[map[]] localhost/logging/logging-fluentd-7zj4m/be5a9f06-22da-11e7-8ac9-0eed4d1f6978:Pod: [localhost/logging/logging-fluentd-7zj4m/be5a9f06-22da-11e7-8ac9-0eed4d1f6978], pod-ip=[172.17.0.4], host-ip=[172.31.19.75], subdomain=[], hostname=[], labels=[map[component:fluentd provider:openshift]], annotations=[map[kubernetes.io/created-by:{"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"DaemonSet","namespace":"logging","name":"logging-fluentd","uid":"be55d462-22da-11e7-8ac9-0eed4d1f6978","apiVersion":"extensions","resourceVersion":"1191"}}
 openshift.io/scc:privileged]], config-map-volumes=[map[config:logging-fluentd]] localhost/openshift-infra/hawkular-metrics-hf6sv/b7f4762f-22da-11e7-8ac9-0eed4d1f6978:Pod: [localhost/openshift-infra/hawkular-metrics-hf6sv/b7f4762f-22da-11e7-8ac9-0eed4d1f6978], pod-ip=[172.17.0.9], host-ip=[172.31.19.75], subdomain=[], hostname=[], labels=[map[metrics-infra:hawkular-metrics name:hawkular-metrics]], annotations=[map[openshift.io/scc:restricted kubernetes.io/created-by:{"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"ReplicationController","namespace":"openshift-infra","name":"hawkular-metrics","uid":"b6e0c0b1-22da-11e7-8ac9-0eed4d1f6978","apiVersion":"v1","resourceVersion":"954"}}
]], config-map-volumes=[map[]]]}
panic: runtime error: index out of range

goroutine 37 [running]:
panic(0x121dd60, 0xc420016070)
	/home/travis/.gimme/versions/go1.7.linux.amd64/src/runtime/panic.go:500 +0x1a1
github.com/hawkular/hawkular-openshift-agent/k8s.K8SEndpoint.GetUrl(0xc4203d55b0, 0x7, 0x13664ab, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/travis/gopath/src/github.com/hawkular/hawkular-openshift-agent/k8s/configmap_entry.go:82 +0x26b
github.com/hawkular/hawkular-openshift-agent/k8s.(*NodeEventConsumer).startCollecting(0xc42046da80, 0xc4203c6260)
	/home/travis/gopath/src/github.com/hawkular/hawkular-openshift-agent/k8s/node_event_consumer.go:169 +0x14f
github.com/hawkular/hawkular-openshift-agent/k8s.(*NodeEventConsumer).consumeNodeEvents(0xc42046da80)
	/home/travis/gopath/src/github.com/hawkular/hawkular-openshift-agent/k8s/node_event_consumer.go:118 +0x26d
created by github.com/hawkular/hawkular-openshift-agent/k8s.(*NodeEventConsumer).Start
	/home/travis/gopath/src/github.com/hawkular/hawkular-openshift-agent/k8s/node_event_consumer.go:85 +0x5c5
```

I am already running `/opt/hawkular/hawkular-openshift-agent` with `-v 5`, but there's no clear indication what the actual problem is here. Thanks for any insights!

**Comment by metmajer** [comment]
@jmazzitelli FYI: I am going to present about monitoring OpenShift apps at the upcoming Red Hat Summit. Let me know if you or any of your peers would be interested in a joint debug session. Otherwise, I'd have to skip HOSA, which I would otherwise like to present in a live demo. Thanks!

**Comment by jmazzitelli** [comment]
@metmajer I've been out on PTO for the past couple days. I'm going to look at this when I get in tomorrow.

In the meantime, start the agent with "-v 5" (to turn on trace logging) in the hopes that that agent logs can provide more details. I'm assuming you are using the latest version of the agent and the latest version of OpenShift? I'm getting the sense that the latest release of OpenShift has broken something in the agent, but that's just a guess (based on feedback I'm seeing in the past week).

**Comment by metmajer** [comment]
@jmazzitelli I have already followed your suggestion using `-v 5` from the other issues I filed recently. And yes, I have been using the latest versions of OpenShift and the agent. Is there any version of OpenShift Origin you would recommend (that is known to work)?

**Comment by jmazzitelli** [comment]
@metmajer I was using v1.5.0-alpha.3 (those are the docker image tags I see). I was building my own from source. But I would say the version prior to the 3.6 release (so I assume use the images that are associated with OpenShift 3.5 would be a good thing to try).

You should hop on #hawkular IRC (on freenode) so we can discuss this in realtime. If need be we can get on the phone. I have a 30-minute meeting in 15 mins and an 60-minute meeting in 2 hours. Free otherwise. But if you get on #hawkular, we might have others in the room that could help as well.

**Comment by metmajer** [comment]
Thanks @jmazzitelli, much appreciated! I'll give that version a try and let you know the outcome. I'll head to the public channel first and bug you only if needed.

**Comment by metmajer** [comment]
The problem was entirely resolved using the latest `1.4.1-Final` tag of the HOSA Docker image.
